src/target/riscv/riscv.c

```diff
@@ -652,6 +654,12 @@ static int add_trigger(struct target *target, struct trigger *trigger)
 			case 6:
 				result = maybe_add_trigger_t6(target, trigger, tdata1);
 				break;
+			case 15:
+				result = maybe_add_trigger_t2(target, trigger, tdata1);
+				if (result == ERROR_OK)
+					break;
+				result = maybe_add_trigger_t6(target, trigger, tdata1);
+				break;
```

Comment:
1) For `tdata1.type == 15`, RISC-V Debug spec says: `15: This trigger exists (...) but is not currently available`. I believe it is more correct for OpenOCD to just ignore such triggers (do not try to use them anyhow). @timsifive - What is your opinion, please?

2) If `tdata1.type` is writable, the supported trigger types should be discovered from `tinfo`. There is a parallel merge request that adds this proper trigger enumeration: https://github.com/riscv/riscv-openocd/pull/725/files @YenHaoChen  - Would it address your use case, too?

Reply:
I thought tdata1.type=15 means the trigger is disabled due to reset or previously configured (writing 0) by the debugger. 

![image](https://user-images.githubusercontent.com/39526191/197714495-45d5e2e6-3654-42aa-91d1-2c679ef289a5.png)

The reset handler writes 0 to the tdata1, which results in tdatat.type=15. Thus, initially, the tdata.type will be 15, and the debugger should try to write the tdata1 if it wants to use a trigger that supports multiple types. Additionally, writing 0 to tdata1 and making tdata1.type=15 are standard procedures for set-up triggers. Thus, I believe a debugger should not ignore a trigger with tdata1.type=15.

![image](https://user-images.githubusercontent.com/39526191/197718640-be22e572-b85a-4f10-a6ba-f278c96a5934.png)

Reply:
> I thought tdata1.type=15 means the trigger is disabled 

You're right - thank you for correcting me.

Still, if the hardware supports multiple trigger types, then it is mandatory to have `tinfo` implemented which provides the information about which types are supported. The parallel merge request https://github.com/riscv/riscv-openocd/pull/725 already implements this discovery.

Reply:
@JanMatCodasip, the spec draft was updated 6 months ago to say this for type 15:
            This trigger is disabled. In this state, \RcsrTdataTwo and
            \RcsrTdataThree can be written with any value that is supported for
            any of the types this trigger implements. The remaining bits in this
            register are ignored.

Seems you're referring to an old draft. You can always see the latest stable draft at https://github.com/riscv/riscv-debug-spec/blob/master/riscv-debug-stable.pdf

Reply:
> Seems you're referring to an old draft.

You're right, I missed this trigger related update in the spec. Thank you correcting me.